test/source/tests/compose.ts

```diff
@@ -916,6 +914,47 @@ export const defineComposeTests = (testVariant: TestVariant, testWithBrowser: Te
       expect(message).to.not.include('Comment');
     }));
 
+    ava.default('compose - multiple compose windows - opening, max 3, order, active', testWithBrowser('ci.tests.gmail', async (t, browser) => {
+      const inboxPage = await browser.newPage(t, TestUrls.extensionInbox('ci.tests.gmail@flowcrypt.test'));
+      // open 3 compose windows
+      await inboxPage.waitAndClick('@action-open-secure-compose-window', { sleepWhenDone: 1 });
+      await inboxPage.waitAndClick('@action-open-secure-compose-window', { sleepWhenDone: 1 });
+      await inboxPage.waitAndClick('@action-open-secure-compose-window', { sleepWhenDone: 1 });
+      let secureComposeWindows = await inboxPage.target.$$('.secure_compose_window');
+      expect(secureComposeWindows.length).to.equal(3);
+      // try to open the 4th one
+      await inboxPage.click('@action-open-secure-compose-window');
+      await inboxPage.waitForContent('.ui-toast-title', 'Only 3 FlowCrypt windows can be opened at a time');
+      // make sure the data-order attributes are correct
+      expect(await PageRecipe.getElementAttribute(secureComposeWindows[0], 'data-order')).to.equal('1');
+      expect(await PageRecipe.getElementAttribute(secureComposeWindows[1], 'data-order')).to.equal('2');
+      expect(await PageRecipe.getElementAttribute(secureComposeWindows[2], 'data-order')).to.equal('3');
+      // make sure the 3rd compose window is active, and the 2nd is previous_active
+      expect(await inboxPage.hasClass('.secure_compose_window[data-order="1"]', 'active')).to.be.false;
+      expect(await inboxPage.hasClass('.secure_compose_window[data-order="2"]', 'active')).to.be.false;
+      expect(await inboxPage.hasClass('.secure_compose_window[data-order="3"]', 'active')).to.be.true;
+      expect(await inboxPage.hasClass('.secure_compose_window[data-order="1"]', 'previous_active')).to.be.false;
+      expect(await inboxPage.hasClass('.secure_compose_window[data-order="2"]', 'previous_active')).to.be.true;
+      expect(await inboxPage.hasClass('.secure_compose_window[data-order="3"]', 'previous_active')).to.be.false;
```

Comment:
If we close window 3 at this stage, I would expect window 2 to become active, and window 1 -- "previous_active".
However, no "previous_active" windows in this case. So I assume, this setup works only in the case of max 3 windows, and will need to be rewritten if we set up max = 4 etc.